### PR TITLE
[Web-App] - Project structure setup 

### PR DESCRIPTION
**Pull request by razoorka**

### CODE REVIEW

File: packages/web/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
```
Comment (alexander2001i) [inline packages/web/.gitignore]: This `.gitignore` is not needed as we have one at the root of the repo. If something that needs to be ignored is not, just move it there

File: packages/web/bitbucket-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+               - npm install
+               - REACT_APP_ENV=stage npm run-script build
+           artifacts:
+                - build/**
+       - step:
+              image: atlassian/pipelines-awscli
+              script:
+               - aws s3 sync --delete build s3://$AWS_BUCKET_TARGET/stage
+               - aws cloudfront create-invalidation --distribution-id $STAGE_DISTRIBUTION_ID  --invalidation-batch "Paths={Quantity=1,Items=['/*']},CallerReference=$(date)"
+              deployment: staging
```
Comment (alexander2001i) [inline packages/web/bitbucket-pipelines.yml]: We use GitHub actions. They are under `.github/workflows`

File: packages/web/package.json [hunk 3]
```diff
@@ -3,31 +3,37 @@
...
   "dependencies": {
-    "@common/types": "^0.0.1",
-    "@testing-library/jest-dom": "^5.11.4",
-    "@testing-library/react": "^11.1.0",
-    "@testing-library/user-event": "^12.1.10",
-    "@types/jest": "^26.0.15",
-    "@types/node": "^12.0.0",
-    "@types/react": "^16.9.53",
-    "@types/react-dom": "^16.9.8",
+    "@welldone-software/why-did-you-render": "^5.0.0",
```
Comment (alexander2001i) [inline packages/web/package.json]: Why is this not in dev dependencies?

File: packages/web/package.json [hunk 4]
```diff
@@ -3,31 +3,37 @@
...
+    "react-redux": "^7.2.2",
+    "react-router-dom": "^5.2.0",
+    "react-scripts": "^3.1.2",
+    "redux": "^4.0.4",
+    "redux-saga": "^1.1.3",
+    "reselect": "^4.0.0",
+    "styled-components": "^5.2.0",
+    "typesafe-actions": "^4.4.2",
+    "yup": "^0.27.0",
+    "jest": "24.9.0"
```
Comment (alexander2001i) [inline packages/web/package.json]: Why is this not in dev dependencies?

File: packages/web/src/containers/Auth/store/saga.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import { AnyAction } from "redux";
+
+import history from "../../../shared/history";
+import { tokenHandler } from "../../../shared/utils";
+import { getUserData, logout, login, registration } from "./actions";
+import api from "../api";
+import { AuthActionTypes } from "./constants";
+import { startLoading, stopLoading } from "../../../shared/store/actions";
+
+function* getUserDataSaga() {
```
Comment (alexander2001i) [inline packages/web/src/containers/Auth/store/saga.tsx:11]: I missed the generators ✨

**Review by alexander2001i — COMMENTED** [review]
Few notes:

* The auth we use is not with email and password, but rather with apple/google signing.
* Why do you use CSS files and not CSS in JS?

**Comment by razoorka** [comment]
> Few notes:
> 
> * The auth we use is not with email and password, but rather with apple/google signing.
> * Why do you use CSS files and not CSS in JS?

1) It is only a boilerplate I will replace all kinds of useless stuff in the nearest future. 
2) I prefer to use sass style instead of styled-components cause it cleaner and does not create additional components nesting.

**Comment by alexander2001i** [comment]
> > Few notes:
> > 
> > * The auth we use is not with email and password, but rather with apple/google signing.
> > * Why do you use CSS files and not CSS in JS?
> 
> 1. It is only a boilerplate I will replace all kinds of useless stuff in the nearest future.
> 2. I prefer to use sass style instead of styled-components cause it cleaner and does not create additional components nesting.

1. Yes, sure. I just mentioned it, no problem at all.
2. I see and I get your point, but I have one question: if we want to share styles between the react native application and the web app how can we do so with sass?

**Comment by razoorka** [comment]
What library would you advise to use for styled-components?

**Review by alexander2001i — APPROVED** [review]
lgtm

**Pull request merged**